apps/gnome-mobile/src/app/(app)/(tabs)/index.tsx

```diff
@@ -101,10 +107,19 @@ const MapScreen = () => {
     };
   }, []);
 
-  let text = "Waiting...";
-  if (errorMsg) {
-    text = errorMsg;
-  }
+  // filtrowanie
+  const filteredGnomes = userLocation
+    ? gnomes.filter((gnome) => {
+        const distance = getDistance(
+          {
+            latitude: userLocation.latitude,
+            longitude: userLocation.longitude,
+          },
+          { latitude: gnome.latitude, longitude: gnome.longitude },
+        );
+        return distance <= maxDistance;
+      })
+    : gnomes;
```

Comment:
tu ogólnie samo ograniczenie wygląda spoko ale przy pierwszym załadowaniu apki czy po włączeniu po prostu kiedy nie ma jeszcze lokalizacji usera to wyświetlą się wszystkie gnomy czego chciałeś pewnie uniknąć

propozycja ode mnie jest taka że skoro jak nie ma lokacji usera to na mapie pokazujemy jakiś punkt we wrocku, to żeby te gnomy początkowo były zaczytane tylko w pobliżu tego obszaru

btw. można ten punkt przenieść defaultowo na środek rynku (ale to brzmi jak osobne zadanie w którym Daniel mógłby zacząć swoje praktyki :D)